Eevee.Sleep.Bot/Controllers/Mongo/TrackedRoleController.cs

```diff
@@ -0,0 +1,30 @@
+using Discord;
+using Eevee.Sleep.Bot.Models;
+using MongoDB.Driver;
+
+namespace Eevee.Sleep.Bot.Controllers.Mongo;
+
+public static class TrackedRoleContoller {
+    public static Task SaveTrackedRole(IRole role) {
+        return MongoConst.TrackedRoleCollection
+            .UpdateOneAsync(
+                Builders<TrackedRoleModel>.Filter.Where(x => x.Id == role.Id),
+                Builders<TrackedRoleModel>.Update.Set(x => x.Name, role.Name),
+                new UpdateOptions { IsUpsert = true }
+            );
+    }
+
+    public static Task RemoveTrackedRole(ulong roleId) {
+        return MongoConst.TrackedRoleCollection
+            .DeleteOneAsync(
+                Builders<TrackedRoleModel>.Filter.Where(x => x.Id == roleId)
+            );
+    }
+
+    public static TrackedRoleModel[] FindAllTrackedRoles() {
```

Comment:
I forgot, but does it have a simple `ToArray()` call, so it doesn't need to be enumerable before changing it to array?

Reply:
Simply returning `IEnumerable` is something we can also do, if it's possible
My main point was to see if it's possible to reduce 2 methods call of `AsEnumerable().ToArray()` to a `ToArray()`.

Reply:
`Find` returns `IFindFluent<TrackedRoleModel, TrackedRoleModel`>, so `ToArray` cannot be called as is
However, I realized that it is easier to handle IEnumerable as it is, so this will be fixed.

Reply:
I'll let you close this one so it serves as your to-do list here :D

Could you also check how it's implemented for any other controller in the repo if there's any for anything returning an array? I just want  to see if the implementation is consistent.

Reply:
For example, this is converted to a HashSet and then returned.

https://github.com/RaenonX-PokemonSleep/eevee-sleep-discord-bot/blob/7a36fb3474478ed17607e497f6a43c65b3cbad87/Eevee.Sleep.Bot/Controllers/Mongo/ActivationPresetController.cs#L8-L16

Reply:
I see, this seems to still be doing `ToEnumerable().To*()` so I guess the 2-step call is required, will mark this as resolved then!